pixels-daemon/src/main/java/io/pixelsdb/pixels/daemon/monitor/util/ScalingManager.java

```diff
@@ -0,0 +1,188 @@
+package io.pixelsdb.pixels.daemon.monitor.util;
+
+import io.pixelsdb.pixels.common.utils.ConfigFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.ec2.Ec2Client;
+import software.amazon.awssdk.services.ec2.model.*;
+
+import java.util.*;
+
+public class ScalingManager {
+    private static final Logger log = LogManager
+            .getLogger(ScalingManager.class);
+    private final Ec2Manager ecManager;
```

Comment:
As if this ScalingManager is implemented only for AWS EC2. However, we have to run scaling on different platforms.